### ISSUE TITLE
Sequence input component gives messages of type "unknown" 

### ISSUE THREAD

**Issue by krazylek** [benthosdev/benthos#1775]
If I use the example provided in documentation:

```yml
input:
  sequence:
    sharded_join:
      type: full-outer
      id_path: uuid
      merge_strategy: array
    inputs:
      - csv:
          paths:
            - ./hobbies.csv
            - ./main.csv
```

The output result is as expected giving me the 3 messages as indicated:

```json
{"age":"28","name":"Emma","uuid":"BBB"}
{"age":"45","hobby":"pokemon go","name":"Geri","uuid":"CCC"}
{"age":"34","hobby":["rowing","golf"],"name":"Melanie","uuid":"AAA"}
```

Unfortunately these messages are not usable in a pipeline, and are of type `"unknown"`.

For example if I add this mapping:

```yml
input:
  sequence:
    sharded_join:
      type: full-outer
      id_path: uuid
      merge_strategy: array
    inputs:
      - csv:
          paths:
            - ./hobbies.csv
            - ./main.csv

pipeline:
  processors:
    - mapping: |
        root = {}
        root.original = this
        root.original_type = this.type()
        root.name_value = this.name
        root.name_type = this.name.type()
```

I get the following output:

```json
{"name_type":"null","name_value":null,"original":{"age":"28","name":"Emma","uuid":"BBB"},"original_type":"unknown"}
{"name_type":"null","name_value":null,"original":{"age":"34","hobby":["rowing","golf"],"name":"Melanie","uuid":"AAA"},"original_type":"unknown"}
{"name_type":"null","name_value":null,"original":{"age":"45","hobby":"pokemon go","name":"Geri","uuid":"CCC"},"original_type":"unknown"}
```

Basically `this` is accessible, with an `unknown` type but none of it's properties are.

Also, any attempt to add a property to `this`will result in an error: 
```yml
    - mapping: |
        root = this
        root.test = 5
```

Gives:
```
failed to assign result (line 2): unable to set target path original as the value of the root was a non-object type (unknown)
```

**Comment by mihaitodor** [comment]
Hey @krazylek, nice find! I proposed a tentative fix in #1776.

**Comment by krazylek** [comment]
Thank you for the prompt fix!
